### ISSUE TITLE
Markdown - utf8 BOM issue - Xed 2.6.0

### ISSUE THREAD

**Issue by artofit** [linuxmint/xed#390]
```
 * Xed version 2.6.0
 * Manjaro 20 XFCE
```

**Steps to reproduce**
1. Have some utf8 markdown file, with BOM unicode.
If you don't have one, just take some .md file, open it with Geany, go in menu document and check "Write the BOM Unicode", save and close the file.

2. Open this file with Xed.
You'll notice that the colours highlighting, that ought to go with markdown such as with #(header1), **(bold), etc. does not work.

3. Close the markdown file.

4. Remove the BOM, for instance open it with Geany: go in menu document and UNCHECK "Write the BOM Unicode", save and close the file.

5. Re-open with Xed
Now the markdown highlighting is as usual, visible.

Attached a markdown file with .txt extension which shows the issue, that you need to change the extension to .md as the upload does not accept .md file, only .txt
[test BOM.txt](https://github.com/linuxmint/xed/files/4917404/test.BOM.txt)

Thanks

**Comment by oldtomdevel** [comment]
Still present in xed 2.6.2 as e.g. in Linux Mint 20.

For clarification I have included two versions of the same example file, one starting with a Byte Order Mark i.e. EF BB BF and one without.

The included example file DemoNoBOM looks like that:

![ScreenshotNoBOM](https://user-images.githubusercontent.com/21215858/95205412-9f830080-07e5-11eb-865d-1f497bc7a3e1.png)

The included example file DemoWithBOM looks like that:

![ScreenshotWithBOM](https://user-images.githubusercontent.com/21215858/95205533-c6413700-07e5-11eb-9e8d-8312a61660c4.png)

However, if you hit F10 (line sorting) and take back that action, it looks as it should...

[DemoNoBOM.cs.txt](https://github.com/linuxmint/xed/files/5333952/DemoNoBOM.cs.txt)

[DemoWithBOM.cs.txt](https://github.com/linuxmint/xed/files/5333966/DemoWithBOM.cs.txt)

Note that the files are originally named `*.cs` and have the mimetype `text/x-csharp`. I just added `.txt` in order to make the upload in Github possible.
